pixy/consumer_test.go

```diff
@@ -0,0 +1,486 @@
+	config.ClientID = "producer"
+	config.Kafka.SeedPeers = testKafkaPeers
+	config.ChannelBufferSize = 1
+	s.producer, err = SpawnGracefulProducer(config)
+	c.Assert(err, IsNil)
+}
+
+func (s *SmartConsumerSuite) TearDownSuite(c *C) {
+	s.producer.Stop()
+}
+
+func (s *SmartConsumerSuite) TestResolveAssignments(c *C) {
+	c.Assert(resolveAssignments(nil, nil), IsNil)
+	c.Assert(resolveAssignments(nil, []string{}), IsNil)
+	c.Assert(resolveAssignments(nil, []string{"a"}), IsNil)
+	c.Assert(resolveAssignments(nil, []string{"a", "b"}), IsNil)
+	c.Assert(resolveAssignments([]int32{}, nil), IsNil)
+	c.Assert(resolveAssignments([]int32{}, []string{}), IsNil)
+	c.Assert(resolveAssignments([]int32{}, []string{"a"}), IsNil)
+	c.Assert(resolveAssignments([]int32{}, []string{"a", "b"}), IsNil)
+	c.Assert(resolveAssignments([]int32{1}, nil), IsNil)
+	c.Assert(resolveAssignments([]int32{1}, []string{}), IsNil)
+
+	c.Assert(resolveAssignments([]int32{0}, []string{"a"}),
+		DeepEquals, map[string]map[int32]none{
+			"a": map[int32]none{0: nothing},
+		})
+	c.Assert(resolveAssignments([]int32{1, 2, 0}, []string{"a"}),
+		DeepEquals, map[string]map[int32]none{
+			"a": map[int32]none{0: nothing, 1: nothing, 2: nothing},
+		})
+	c.Assert(resolveAssignments([]int32{0}, []string{"b", "a"}),
+		DeepEquals, map[string]map[int32]none{
+			"a": map[int32]none{0: nothing},
+		})
+	c.Assert(resolveAssignments([]int32{0, 3, 1, 2}, []string{"b", "a"}),
+		DeepEquals, map[string]map[int32]none{
+			"a": map[int32]none{0: nothing, 1: nothing},
+			"b": map[int32]none{2: nothing, 3: nothing},
+		})
+	c.Assert(resolveAssignments([]int32{0, 3, 1, 2}, []string{"b", "c", "a"}),
+		DeepEquals, map[string]map[int32]none{
+			"a": map[int32]none{0: nothing, 1: nothing},
+			"b": map[int32]none{2: nothing},
+			"c": map[int32]none{3: nothing},
+		})
+	c.Assert(resolveAssignments([]int32{0, 3, 1, 2, 4}, []string{"b", "c", "a"}),
+		DeepEquals, map[string]map[int32]none{
+			"a": map[int32]none{0: nothing, 1: nothing},
+			"b": map[int32]none{2: nothing, 3: nothing},
+			"c": map[int32]none{4: nothing},
+		})
+	c.Assert(resolveAssignments([]int32{0, 3, 1, 2, 5, 4}, []string{"b", "c", "a"}),
+		DeepEquals, map[string]map[int32]none{
+			"a": map[int32]none{0: nothing, 1: nothing},
+			"b": map[int32]none{2: nothing, 3: nothing},
+			"c": map[int32]none{4: nothing, 5: nothing},
+		})
+	c.Assert(resolveAssignments([]int32{6, 0, 3, 1, 2, 5, 4}, []string{"b", "c", "a"}),
+		DeepEquals, map[string]map[int32]none{
+			"a": map[int32]none{0: nothing, 1: nothing, 2: nothing},
+			"b": map[int32]none{3: nothing, 4: nothing},
+			"c": map[int32]none{5: nothing, 6: nothing},
+		})
+	c.Assert(resolveAssignments([]int32{6, 0, 3, 1, 2, 5, 4}, []string{"d", "b", "c", "a"}),
+		DeepEquals, map[string]map[int32]none{
+			"a": map[int32]none{0: nothing, 1: nothing},
+			"b": map[int32]none{2: nothing, 3: nothing},
+			"c": map[int32]none{4: nothing, 5: nothing},
+			"d": map[int32]none{6: nothing},
+		})
```

Comment:
Also, you don't need to write out the full map type name here. If you run `gofmt -s -w consumer_test.go` it will automatically simplify this for you. If you have gofmt-on-save for your editor, you can add the "-s" flag to always do simplifications like this.